analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Binary.cs

```diff
@@ -657,4 +657,44 @@ public void Binary_StringConcatenation_Compund_CS(string expression)
         validator.ValidateContainsOperation(OperationKind.Binary);
         validator.ValidateTag("S", x => x.Should().HaveOnlyConstraint(ObjectConstraint.Null)); // FIXME: s is not null here (https://github.com/SonarSource/sonar-dotnet/issues/7111)
     }
+
+    [DataTestMethod]
+    [DataRow("i + j", 47)]
+    [DataRow("i + 1", 43)]
+    [DataRow("i + j + 1", 48)]
+    [DataRow("(i + 1) + j", 48)]
+    [DataRow("(i + j) + 1", 48)]
+    [DataRow("i + (1 + j)", 48)]
+    [DataRow("1 + (i + j)", 48)]
+    [DataRow("i - j", 37)]
+    [DataRow("j - i", -37)]
+    [DataRow("i - 1", 41)]
+    [DataRow("i - j - 1", 36)]
+    [DataRow("(i - 1) - j", 36)]
+    [DataRow("(i - j) - 1", 36)]
+    [DataRow("i - (j - 1)", 38)]
+    [DataRow("i - (1 - j)", 46)]
+    [DataRow("1 - (i - j)", -36)]
+    [DataRow("i + j - 1", 46)]
+    [DataRow("(i + j) - 1", 46)]
+    [DataRow("(i + 1) - j", 38)]
+    [DataRow("i + (j - 1)", 46)]
+    [DataRow("i + (1 - j)", 38)]
+    [DataRow("1 + (i - j)", 38)]
+    [DataRow("i - j + 1", 38)]
+    [DataRow("(i - j) + 1", 38)]
+    [DataRow("(i - 1) + j", 46)]
+    [DataRow("i - (j + 1)", 36)]
+    [DataRow("1 - (i + j)", -46)]
+    public void Binary_Calculations_UpdatesNumberConstraint(string expression, int result)
```

Comment:
To make the role in UT more obvious
```suggestion
    public void Binary_Calculations_UpdatesNumberConstraint(string expression, int expected)
```